internal/app/machined/pkg/controllers/network/address_event.go

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"context"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"go.uber.org/zap"
+
+	"github.com/talos-systems/talos/internal/app/machined/pkg/runtime"
+	"github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/resources/k8s"
+	"github.com/talos-systems/talos/pkg/machinery/resources/network"
+)
+
+// AddressEventController reports aggregated enpoints state from hostname statuses and k8s endpoints
+// to the events stream.
+type AddressEventController struct {
+	V1Alpha1Events runtime.Publisher
+}
+
+// Name implements controller.Controller interface.
+func (ctrl *AddressEventController) Name() string {
+	return "network.AddressEventController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *AddressEventController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: network.NamespaceName,
+			Type:      network.NodeAddressType,
+			Kind:      controller.InputWeak,
+			ID: pointer.ToString(network.FilteredNodeAddressID(
+				network.NodeAddressCurrentID,
+				k8s.NodeAddressFilterNoK8s)),
+		},
+		{
+			Namespace: network.NamespaceName,
+			Type:      network.HostnameStatusType,
+			Kind:      controller.InputWeak,
+			ID:        pointer.ToString(network.HostnameID),
+		},
+	}
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *AddressEventController) Outputs() []controller.Output {
+	return nil
+}
+
+// Run implements controller.Controller interface.
+func (ctrl *AddressEventController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
+	ticker := time.NewTicker(time.Minute * 10)
+
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+		case <-r.EventCh():
```

Comment:
feels like `select` should end here with `}` to make it act both on ticker and reconcile events